lib/screens/pdf/pdfmain.dart

```diff
@@ -0,0 +1,134 @@
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:path_provider/path_provider.dart';
+
+import 'package:pdf/pdf.dart';
+import 'package:pdf/widgets.dart' as pdf;
+import 'package:printing/printing.dart';
+
+import 'document.dart';
+import 'viewer.dart';
+
+class PdfDemo extends StatefulWidget {
+  @override
+  PdfDemoState createState() {
+    return PdfDemoState();
+  }
+}
+
+class PdfDemoState extends State<PdfDemo> {
+  final GlobalKey<State<StatefulWidget>> shareWidget = GlobalKey();
+  final GlobalKey<State<StatefulWidget>> previewContainer = GlobalKey();
+
+  Future<void> _printPdf() async {
+    print('Print ...');
+    await Printing.layoutPdf(
+        onLayout: (PdfPageFormat format) async =>
+            (await generateDocument(format)).save());
+  }
+
+  Future<void> _saveAsFile() async {
+    final Directory appDocDir = await getApplicationDocumentsDirectory();
+    final String appDocPath = appDocDir.path;
+    final File file = File(appDocPath + '/' + 'document.pdf');
+    print('Save as file ${file.path} ...');
+    await file.writeAsBytes((await generateDocument(PdfPageFormat.a3)).save());
+    Navigator.push<dynamic>(
+      context,
+      MaterialPageRoute<dynamic>(
+          builder: (BuildContext context) => PdfViewer(file: file)),
+    );
+  }
+
+  Future<void> _sharePdf() async {
+    print('Share ...');
+    final pdf.Document document = await generateDocument(PdfPageFormat.a3);
+
+    // Calculate the widget center for iPad sharing popup position
+    final RenderBox referenceBox =
+    shareWidget.currentContext.findRenderObject();
+    final Offset topLeft =
+    referenceBox.localToGlobal(referenceBox.paintBounds.topLeft);
+    final Offset bottomRight =
+    referenceBox.localToGlobal(referenceBox.paintBounds.bottomRight);
+    final Rect bounds = Rect.fromPoints(topLeft, bottomRight);
+
+    await Printing.sharePdf(
+        bytes: document.save(), filename: 'my-résumé.pdf', bounds: bounds);
+  }
+
+  Future<void> _printScreen() async {
+    final RenderRepaintBoundary boundary =
+    previewContainer.currentContext.findRenderObject();
+    final ui.Image im = await boundary.toImage();
+    final ByteData bytes =
+    await im.toByteData(format: ui.ImageByteFormat.rawRgba);
+    print('Print Screen ${im.width}x${im.height} ...');
+
+    Printing.layoutPdf(onLayout: (PdfPageFormat format) {
+      final pdf.Document document = PdfDoc();
+
+      final PdfImage image = PdfImage(document.document,
+          image: bytes.buffer.asUint8List(),
+          width: im.width,
+          height: im.height);
+
+      document.addPage(pdf.Page(
+          pageFormat: format,
+          build: (pdf.Context context) {
+            return pdf.Center(
+              child: pdf.Expanded(
+                child: pdf.Image(image),
+              ),
+            ); // Center
+          })); // Page
+
+      return document.save();
+    });
+  }
+
+  Future<void> _printHtml() async {
```

Comment:
We've removed it since we've generated PDF using pure dart so this shouldn't be a problem anymore. You can check updated file here: https://github.com/hkis-gh/appsolidariav2/blob/pdf_task/lib/screens/pdf/pdfmain.dart